icecap/crates

```diff
@@ -0,0 +1 @@
+icecap/src/rust/crates
```

Comment:
It took me a while to work out that this is a symbolic link because GitHub doesn't exactly make it clear. There are so many directories called `crates`, and at least three called `icecap`, and I don't know how many called `src` or `rust`, so I really can't get my bearings here. What is this for?

Reply:
There are a lot of `crates` symbolic links because of the workspace requirement to contain only sub-folders. We use symbolic links to work around that and share crates among workspaces. The `crates` symbolic link in each workspace points to the top-level repository directory to access the different rust crates of the project. Eventually @nspin and I have suggested moving all the Veracruz crates into a `crates` directory (and update the symbolic links).

The current `icecap` build system has some assumptions about the topology so it was easier to add `icecap` symbolic links to `icecap/icecap` to have both the current and the workspaces build systems work in parallel.

As for `icecap/crates` linking to `icecap/icecap/src/rust`, it was a way to reduce some paths in `Cargo.toml` (e.g. `runtime-manager`),  turning `../icecap/icecap/src/rust/icecap/icecap-core` into `../icecap/crates/icecap/icecap-core`.

Once we switch to workspaces there will be opportunities to simplify some of this.